Recrutify/Recrutify.Services/Services/CandidateService.cs

```diff
@@ -46,10 +56,28 @@ public async Task<CandidateDTO> CreateAsync(CandidateCreateDTO candidateCreateDT
             return result;
         }
 
-        public Task UpsertFeedbackAsync(Guid id, Guid projectId, CreateFeedbackDTO feedbackDto)
+        public async Task UpsertFeedbackAsync(Guid id, Guid projectId, CreateFeedbackDTO feedbackDto)
         {
-            var feedback = _mapper.Map<Feedback>(feedbackDto);
-            return _candidateRepository.UpsertFeedbackAsync(id, projectId, feedback);
+            var candidateWithProjectFeedback = await _candidateRepository.GetCandidateWithProjectFeedbackAsync(id, projectId,
+                feedbackDto.UserId, _mapper.Map<FeedbackType>(feedbackDto.Type));
+            if (candidateWithProjectFeedback == null)
+            {
+                var newFeedback = _mapper.Map<Feedback>(feedbackDto);
+                newFeedback.CreatedOn = DateTime.UtcNow;
+                await _candidateRepository.UpsertFeedbackAsync(id, projectId, newFeedback);
+            }
+            else
+            {
+                var copy = candidateWithProjectFeedback.DeepCopy();
+                var feedbackToUpdate = _mapper.Map(feedbackDto, copy);
```

Comment:
pass candidateWithProjectFeedback.DeepCopy() in Map method, don't declare var copy

Reply:
done